Code/Source/solver/set_bc.cpp

```diff
@@ -228,11 +236,18 @@ void calc_der_cpl_bc(ComMod& com_mod, const CmMod& cm_mod)
         // Finite difference calculation of the resistance dP/dQ
         bc.r = (cplBC.fa[i].y - orgY[i]) / diff;
 
+        // Set resistance for capping BC if it exists for this BC.
+        // Set it to be the same resistance as the capped BC.
+        auto& iCapBC = bc.iCapBC;
+        if (iCapBC != -1) {
```

Comment:
Once again we have an implicit meaning for -1 which is set somewhere. It is much better to have bool member data to this so then you can write 
```
if (bc.is_coupling_cap) {
}
```
and you don't need a comment and you can search to see who sets this.